architecture/ambient/ztunnel.md

```diff
@@ -0,0 +1,236 @@
+# Ztunnel
+
+This document provides an overview of the architecture and design decisions around Ztunnel, the node-proxy component in ambient mode.
+
+## Background and motivation
```

Comment:
I think it may help new folks if there was a bit more of a description of the ambient architecture, so they know where the ztunnel fits.  This section motivates some of the reasoning, but when we get to the Goals section, I don't think it will be clear to new folks how it fits in.  I'm happy to suggest some edits or do a follow-on PR, if it would help.